core-processor/src/executor.rs

```diff
@@ -321,7 +320,7 @@ pub fn execute_wasm<A: ProcessorExt + EnvExt + IntoExtInfo + 'static, E: Environ
         }
 
         if let Some(initial_data) = initial_pages.get(&page) {
-            let old_data = initial_data.as_ref().ok_or_else(|| ExecutionError {
+            let old_data = initial_data.as_ref().ok_or(ExecutionError {
```

Comment:
why clone without need? what wrong with or_else lazy block?

Reply:
both `ok_or` and `ok_or_else` accept closures as arguments that move inside `info.gas_amount`. Even with try operator (`?`) rust doesn't understand that this move is legit. I changed the code to have more explicit control flow.